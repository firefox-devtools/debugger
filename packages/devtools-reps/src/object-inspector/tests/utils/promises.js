/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const {
  makeNodesForPromiseProperties,
  nodeIsPromise,
} = require("../../utils/node");

describe("promises utils function", () => {
  it("is promise", () => {
    const promise = {
      contents: {
        enumerable: true,
        configurable: false,
        value: {
          actor: "server2.conn2.child1/pausedobj36",
          promiseState: {
            state: "rejected",
            reason: {
              type: "undefined"
            },
          },
          class: "Promise",
          type: "object",
        },
      }
    };

    expect(nodeIsPromise(promise)).toEqual(true);
  });

  it("makeNodesForPromiseProperties", () => {
    const promise = {
      path: "root",
      contents: {
        value: {
          actor: "server2.conn2.child1/pausedobj36",
          promiseState: {
            state: "rejected",
            reason: {
              type: "3"
            },
          },
          class: "Promise",
          type: "object",
        },
      }
    };

    const properties = makeNodesForPromiseProperties(promise);
    expect(properties).toMatchSnapshot();
  });
});
