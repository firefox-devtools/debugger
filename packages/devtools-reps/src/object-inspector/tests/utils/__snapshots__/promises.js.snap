// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promises utils function makeNodesForPromiseProperties 1`] = `
Array [
  Object {
    "contents": Object {
      "value": "rejected",
    },
    "meta": undefined,
    "name": "<state>",
    "parent": Object {
      "contents": Object {
        "value": Object {
          "actor": "server2.conn2.child1/pausedobj36",
          "class": "Promise",
          "promiseState": Object {
            "reason": Object {
              "type": "3",
            },
            "state": "rejected",
          },
          "type": "object",
        },
      },
      "path": "root",
    },
    "path": Symbol(root/<state>),
    "type": Symbol(<state>),
  },
  Object {
    "contents": Object {
      "value": Object {
        "type": "3",
      },
    },
    "meta": undefined,
    "name": "<reason>",
    "parent": Object {
      "contents": Object {
        "value": Object {
          "actor": "server2.conn2.child1/pausedobj36",
          "class": "Promise",
          "promiseState": Object {
            "reason": Object {
              "type": "3",
            },
            "state": "rejected",
          },
          "type": "object",
        },
      },
      "path": "root",
    },
    "path": Symbol(root/<reason>),
    "type": Symbol(<reason>),
  },
]
`;
